### PR TITLE
UI: gnosis staking

### PR DESCRIPTION
**Pull request by kangarang**
this feature pr adds support for gnosis staking. included are style adjustments, code cleanup, and minimal changes to the hooks and calculations to support multiple chains and network switching.

matic staking (left image shows the same calculated results as the current live site):

Screen Shot 2022-02-01 at 1 22 40 PM

gnosis staking:

Screen Shot 2022-02-01 at 1 15 47 PM

### CODE REVIEW

File: packages/frontend/src/utils/networks.ts [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+import { Signer, providers } from 'ethers'
 import { find } from 'lodash'
 import { networks } from 'src/config'
 import { allNetworks } from 'src/config/networks'
 import Network from 'src/models/Network'
 
-export function findNetworkBySlug(slug: string) {
-  return find(allNetworks, ['slug', slug])
+export function findNetworkBySlug(slug: string, networks?: Network[]) {
+  return find(networks || allNetworks, ['slug', slug])
```
Comment (miguelmota) [inline packages/frontend/src/utils/networks.ts:10]: Assigning default in method signature would be cleaner

```typescript
export function findNetworkBySlug(slug: string, networks: Network[] = allNetworks) {
  return find(networks, ['slug', slug])
}
```

File: packages/frontend/src/pages/Stake/Stake.tsx [hunk 2]
```diff
@@ -1,115 +1,148 @@
...
+import { findMatchingBridge, findNetworkBySlug, getNetworkProviderOrDefault } from 'src/utils'
 import { StakingRewards__factory } from '@hop-protocol/core/contracts'
+import { providers, Signer } from 'ethers'
+import Network from 'src/models/Network'
+import { RaisedNetworkSelector } from 'src/components/NetworkSelector/RaisedNetworkSelector'
 
-const useStyles = makeStyles(theme => ({
-  container: {
-    padding: `${theme.padding.default} 0`,
+const stakingRewardsContracts = {
```
Comment (miguelmota) [inline packages/frontend/src/pages/Stake/Stake.tsx:18]: minor suggestion:

Move the addresses config to the config dir

File: packages/frontend/src/pages/Stake/Stake.tsx [hunk 3]
```diff
@@ -1,115 +1,148 @@
...
   const { provider } = useWeb3Context()
 
-  const stakingContracts = {
-    USDC: '0x2C2Ab81Cf235e86374468b387e241DF22459A265',
-    USDT: '0x07932e9A5AB8800922B2688FB1FA0DAAd8341772',
-    MATIC: '0x7dEEbCaD1416110022F444B03aEb1D20eB4Ea53f',
-    ETH: '0x7bceda1db99d64f25efa279bb11ce48e15fda427',
-  }
+  const availableNetworks = networks.filter(n =>
+    [ChainSlug.Gnosis, ChainSlug.Polygon].includes(n.slug as ChainSlug)
```
Comment (miguelmota) [inline packages/frontend/src/pages/Stake/Stake.tsx:42]: maybe these chain slugs can be dynamic based on the staking addresses config keys?

File: packages/frontend/src/pages/Stake/Stake.tsx [hunk 4]
```diff
@@ -1,115 +1,148 @@
...
+  // GNO
+  const daiBridge = useAsyncMemo(async () => findMatchingBridge(bridges, 'DAI'), [bridges])
+  const daiStakingToken = useAsyncMemo(async () => {
+    return daiBridge?.getSaddleLpToken(ChainSlug.Gnosis)
+  }, [daiBridge])
+  const gnosisStakingRewards = useAsyncMemo(async () => {
+    const gnosisProvider = await sdk.getSignerOrProvider(ChainSlug.Gnosis)
+    const _provider = getNetworkProviderOrDefault(ChainId.Gnosis, gnosisProvider, provider)
+    return StakingRewards__factory.connect(stakingRewardsContracts.gnosis.DAI, _provider)
+  }, [sdk, provider, user])
```
Comment (miguelmota) [inline packages/frontend/src/pages/Stake/Stake.tsx:100]: I don't think the `user` is being used here

File: packages/frontend/src/utils/networks.ts [hunk 5]
```diff
@@ -69,3 +71,26 @@ export function isL2ToL1(srcNetwork: Network, destNetwork: Network) {
 
   return false
 }
+
+export function isProviderNetworkByChainSlugOrId(
```
Comment (miguelmota) [inline packages/frontend/src/utils/networks.ts:75]: suggestion:

Remove the slug type and just use chainId for both of this methods

**Comment by miguelmota** [comment]
this query debugger is really cool. It won't render on live site right?

DeepinScreenshot_select-area_20220202101948

File: packages/frontend/src/pages/Stake/Stake.tsx (hunk 3 above)
Comment (kangarang) [reply packages/frontend/src/pages/Stake/Stake.tsx:42]: 👍

```js
const availableNetworks = networks.filter(n =>
  Object.keys(stakingRewardsContracts).includes(n.slug as ChainSlug)
)
```

**Comment by kangarang** [comment]
> this query debugger is really cool. It won't render on live site right?

@miguelmota yep, it's not compiled for production builds

**Pull request merged**